[Bug]: conn.RemoteAddr().String() panic

### Actions I've taken before I'm here

- [x] I've thoroughly read the documentations about this problem but still have no answer.
- [x] I've searched the Github Issues/Discussions but didn't find any similar problems that have been solved.
- [x] I've searched the internet for this problem but didn't find anything helpful.

### Questions with details

gnet 版本： v2.8.0
操作系统： Linux 5.4.241-19-0017.1_plus x86_64
go版本：go 1.20
相关日志：
xxxx/server/gnet.(*GNetServer).OnTraffic(0xc000be13e0?, {0xc3d020, 0xc0002d1960})
	/builds/xxx/inr.go:96 +0x77
xxxx/biz.(*InrHandler).Handle(0xc0001dd080?, {0xc347c8, 0xc000122000}, {0x7f5bb42c5510, 0xc0002d1960}, 0xc01c660180)
	/builds/xxxx/biz/inr.go:178 +0x48b
xxxx/biz.(*InrHandler).heartbeat(0xc347c8?, {0xc347c8, 0xc000122000}, {0x7f5bb42c5510, 0xc0002d1960}, 0xc01c660180)
goroutine 75 [running]:
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x9e192b]
panic: runtime error: invalid memory address or nil pointer dereference

问题：
1. 出现panic的原因是什么？
2. panic以后gnet就挂了，有什么方法可以自动恢复吗？
### Code snippets (optional)

```go

func (ss *GNetServer) OnTraffic(c gnet.Conn) (action gnet.Action) {
	for c.InboundBuffered() > 0 {
		msg, err := h.Decode(ctx, c)
		if err != nil {
			if errors.Is(err, io.ErrShortBuffer) {
				return
			}
			return gnet.Close
		}
		_ = h.Handle(ctx, c, msg)
	}

	if c.InboundBuffered() > 0 {
		if err := c.Wake(nil); err != nil { // wake up the connection manually to avoid missing the leftover data
			fls.Error(ctx, "failed to wake up the connection", fls.ErrorField(err))
			return gnet.Close
		}
	}
	return
}

func (h *InrHandler) Handle(ctx context.Context, conn net.Conn, msg *codec.CmnOMessage) error {
		...
		h.heartbeat(ctx, conn, msg)
		...

}

func (h *InrHandler) heartbeat(ctx context.Context, conn net.Conn, msg *codec.CmnOMessage) error {
	...
	if _, err := conn.Write(data); err != nil {
		return errors.Wrap(err, "write heartbeat failed")
	}
	log.Debug(ctx, "handle heartbeat success", fls.String("remote_addr", conn.RemoteAddr().String())) // panic line
	...
}

```

--- a/connection_unix.go
+++ b/connection_unix.go
@@ -148,6 +148,12 @@ func (c *conn) write(data []byte) (n int, err error) {
 		return
 	}
 
+	defer func() {
+		if err != nil {
+			_ = c.loop.close(c, os.NewSyscallError("write", err))
+		}
+	}()
+
 	var sent int
 loop:
 	if sent, err = unix.Write(c.fd, data); err != nil {
@@ -160,7 +166,7 @@ loop:
 			}
 			return
 		}
-		return 0, c.loop.close(c, os.NewSyscallError("write", err))
+		return 0, err
 	}
 	data = data[sent:]
 	if isET && len(data) > 0 {
@@ -191,6 +197,12 @@ func (c *conn) writev(bs [][]byte) (n int, err error) {
 		return
 	}
 
+	defer func() {
+		if err != nil {
+			_ = c.loop.close(c, os.NewSyscallError("writev", err))
+		}
+	}()
+
 	remaining := n
 	var sent int
 loop:
@@ -204,7 +216,7 @@ loop:
 			}
 			return
 		}
-		return 0, c.loop.close(c, os.NewSyscallError("writev", err))
+		return 0, err
 	}
 	pos := len(bs)
 	if remaining -= sent; remaining > 0 {

--- a/connection_unix.go
+++ b/connection_unix.go
@@ -162,7 +162,7 @@ loop:
 		if err == unix.EAGAIN {
 			_, err = c.outboundBuffer.Write(data)
 			if !isET {
-				err = c.loop.poller.ModReadWrite(&c.pollAttachment, false)
+				err = c.loop.poller.ModReadWrite(&c.pollAttachment, isET)
 			}
 			return
 		}
@@ -175,7 +175,7 @@ loop:
 	// Failed to send all data back to the remote, buffer the leftover data for the next round.
 	if len(data) > 0 {
 		_, _ = c.outboundBuffer.Write(data)
-		err = c.loop.poller.ModReadWrite(&c.pollAttachment, false)
+		err = c.loop.poller.ModReadWrite(&c.pollAttachment, isET)
 	}
 
 	return
@@ -212,7 +212,7 @@ loop:
 		if err == unix.EAGAIN {
 			_, err = c.outboundBuffer.Writev(bs)
 			if !isET {
-				err = c.loop.poller.ModReadWrite(&c.pollAttachment, false)
+				err = c.loop.poller.ModReadWrite(&c.pollAttachment, isET)
 			}
 			return
 		}
@@ -238,7 +238,7 @@ loop:
 	// Failed to send all data back to the remote, buffer the leftover data for the next round.
 	if remaining > 0 {
 		_, _ = c.outboundBuffer.Writev(bs)
-		err = c.loop.poller.ModReadWrite(&c.pollAttachment, false)
+		err = c.loop.poller.ModReadWrite(&c.pollAttachment, isET)
 	}
 
 	return

